### DISCUSSION TITLE
Specifying type of columns created from capture groups (';' command)?

### DISCUSSION THREAD

**Discussion by mhoban** [saulpw/visidata#2310] [Q&A] [answered]
I have a dataset with a first column that consists of a character string which is a word with some numbers after it (e.g., 'item1', 'item77', 'item1073'). In the interest of sorting the column by the value of those numbers rather than alphabetically, I thought I'd create a new column from just the number part, so I did this:

```
; (\d+)
```

That seemed to work just fine. But actually it seems to have returned the tuple/list of capture groups rather than the specific capture group itself? Or at least that's how I'm interpreting it, because the contents of the new column show the index as well as the numeric value, like this:

```
[1] 1063
[1] 1071
[1] 1138
...
```

And when I try to type it as an integer (with the '#' command), the cell contents start to look like this:

```
('1063',)!
('1071',)!
('1138',)!
...
```
(those exclamation points are all red)

Am I missing something about the capture command? How can I get *just* the value of the capture group, so I can call it an integer and sort it?

Thanks!

**Comment by saulpw** [comment, marked as answer]
Hi @mhoban, yes, it's a list column since v3.0. Use the `(` (`expand-col`) command to expand the column into its constituent elements.

**Reply by mhoban** [reply]
This did it. Thanks!
